### PR TITLE
Functional tests : Shopping Carts - delete abandoned carts in BO

### PR DESCRIPTION
**Pull request by cfarhani06**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  develop
| Description?      | BO - delete abandoned carts
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+// Import BO pages
+const dashboardPage = require('@pages/BO/dashboard');
+const shoppingCartsPage = require('@pages/BO/orders/shoppingCarts');
+
+// Import data
+const {DefaultCustomer} = require('@data/demo/customer');
+const {Products} = require('@data/demo/products');
+
+const baseContext = 'functional_BO_Shopping_Carts_Delete_Abandoned_Carts';
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]:
```suggestion
const baseContext = 'functional_BO_orders_shoppingCarts_deleteAbandonedCarts';
```

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+- creat shopping cart from FO
+Scenario:
+- Filter shopping carts by non ordered
+- delete them if exist
+ */
+
+// Pre-condition: Create 1 order in FO
+createShoppingCart(orderByCustomerData, `${baseContext}_preTest_1`);
+
+describe('BO - Orders : delete abandoned carts', async () => {
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]:
```suggestion
describe('BO - Orders : Delete abandoned carts', async () => {
```

**Review by Progi1984 — COMMENTED** [review]
Small feedbacks

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 1 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 2 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+/*
+Pre-condition:
+- creat shopping cart from FO
+Scenario:
+- Filter shopping carts by non ordered
+- delete them if exist
+ */
+
+// Pre-condition: Create 1 order in FO
+createShoppingCart(orderByCustomerData, `${baseContext}_preTest_1`);
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: The pre-condition should be inside the describe

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  });
+
+  it('should reset all filters and get number of shopping carts', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'resetFiltersFirst', baseContext);
+
+    numberOfShoppingCarts = await shoppingCartsPage.resetAndGetNumberOfLines(page);
+    await expect(numberOfShoppingCarts).to.be.above(0);
+  });
+
+  it('should search the non ordered shopping carts and delete them if exist', async function () {
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]:
```suggestion
  it('should search the non ordered shopping carts', async function () {
```

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    const numberOfShoppingCartsAfterFilter = await shoppingCartsPage.getNumberOfElementInGrid(page);
+    await expect(numberOfShoppingCartsAfterFilter).to.be.at.most(numberOfShoppingCarts);
+
+    numberOfShoppingCarts -= numberOfShoppingCartsAfterFilter;
+
+    for (let row = 1; row <= numberOfShoppingCartsAfterFilter; row++) {
+      const textColumn = await shoppingCartsPage.getTextColumn(page, row, 'c!lastname');
+      await expect(textColumn).to.contains('Non ordered');
+    }
+
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js:100]:
```suggestion
it('should delete the non ordered shopping carts', async function () {
```

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  product: Products.demo_1,
+  productQuantity: 1,
+};
+
+/*
+Pre-condition:
+- creat shopping cart from FO
+Scenario:
+- Filter shopping carts by non ordered
+- delete them if exist
```
Comment (nesrineabdmouleh) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]:
```suggestion
- Delete them
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@cfarhani06 Ola, you can run commands `GENERATE_FAILED_STEPS=true npm run test:all` which generate the file `mochawesome.json` and `npm run check:step-identifiers` for detecting errors in step identifiers.

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
+require('module-alias/register');
+
+// Import expect from chai
+const {expect} = require('chai');
+
+// Import utils
+import helper from '@utils/helpers';
+
+// Import test context
+import testContext from '@utils/testContext';
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]:
```suggestion
// Import utils
import helper from '@utils/helpers';
import testContext from '@utils/testContext';

require('module-alias/register');

// Import expect from chai
const {expect} = require('chai');
```

you should move it before requires

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 3 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 4 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 6 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 7 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js]: Done

File: package-lock.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (Progi1984) [inline package-lock.json]: Please remove this file ;)

Comment (cfarhani06) [reply package-lock.json]: Done

File: tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js (hunk 5 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/02_orders/05_shoppingCarts/05_deleteAbandonedCarts.js:100]: Done

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by SD1982 — APPROVED** [review]

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by matks** [comment]
Thank you @cfarhani06

**Pull request merged**